tests/utils.py

```diff
@@ -1,7 +1,27 @@
+import contextlib
 import platform
+import socket
+import subprocess
+import sys
 
 import pytest
 
 skip_on_windows = pytest.mark.skipif(
     platform.system() == "Windows", reason="Test is broken skipped on Windows"
 )
+
+
+@contextlib.contextmanager
+def background_subprocess(*args, stdout=sys.stdout, stderr=sys.stdout, **kwargs):
```

Comment:
We had this before, but it was removed in #322. Probably can also be used by the UI tests, but we can do that in a follow-up.